Classes/Issues/GithubClient+Issues.swift

```diff
@@ -269,4 +269,52 @@ extension GithubClient {
         }))
     }
 
+    func fetchViewerCollaborator(
+        owner: String,
+        repo: String,
+        completion: @escaping (Result<Bool>) -> Void
+        ) {
+        guard let viewer = userSession?.username else {
+            completion(.error(nil))
+            return
+        }
+
+        // https://developer.github.com/v3/repos/collaborators/#check-if-a-user-is-a-collaborator
+        request(Request(
+            path: "repos/\(owner)/\(repo)/collaborators/\(viewer)",
```

Comment:
We do everything in our power to prevent caching these requests, but should we add something so we don't constantly hit this endpoint? Thinking of our rate limits!

Reply:
I was thinking about caching this too. I think you're right, this is worth following up w/ maybe a singleton that caches your collab state in memory for the lifetime of the process. If you were removed as a collab we just return an error result. How's that sound?

Reply:
Sounds good, if we get an error result later in the flow then we just clear the cache for that repo so it updates?